### PR TITLE
Bug 2104402: Merge wizard and vm Environment editors

### PR DESCRIPTION
**Pull request by upalatucci**
## 📝 Description

Please guys don't be scared by the # of files edited. 
Most of them get deleted.

Merging environment editors
Creating EnvironmentForm to make an easier interface

follow-up pr will actually fix the bug

### CODE REVIEW

File: src/utils/components/EnvironmentEditor/EnvironmentForm.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    onReload,
+    onSave,
+    edited,
+    error: formError,
+    setError,
+  } = useEnvironments(vm, updateVM, onEditChange);
+
+  const environmentNamesSelected = environments.map((env) => env.name);
+
+  const closeError = () => {
```
Comment (metalice) [inline src/utils/components/EnvironmentEditor/EnvironmentForm.tsx:47]: i think closeError is used only once? you can simply send ()=>setError(null) , no need to declare

File: src/utils/components/EnvironmentEditor/components/EnvironmentFormActions.tsx [hunk 2]
```diff
@@ -29,15 +29,19 @@ const WizardEnvironmentTabFooter: React.FC<WizardEnvironmentTabFooterProps> = ({
 }) => {
   const { t } = useKubevirtTranslation();
   const [success, setSuccess] = React.useState(false);
+  const [loading, setLoading] = React.useState(false);
   const [apiError, setApiError] = React.useState<any>();
 
   const _onSave = async () => {
```
Comment (metalice) [inline src/utils/components/EnvironmentEditor/components/EnvironmentFormActions.tsx:35]: why starting with _ ?

File: src/views/catalog/wizard/tabs/environment/WizardEnvironmentTab.tsx [hunk 3]
```diff
@@ -1,107 +1,28 @@
...
-      </div>
-
-      <WizardEnvironmentTabFooter
-        error={loadError || formError}
-        onSave={onSave}
-        onReload={onReload}
-        closeError={closeError}
-        isSaveDisabled={!edited || !environments.every((env) => env.name)}
-      />
+      <EnvironmentForm vm={vm} onEditChange={onEditChange} updateVM={updateVM} />
```
Comment (metalice) [inline src/views/catalog/wizard/tabs/environment/WizardEnvironmentTab.tsx:25]: i think if you are using once you can simply use (edited) => setDisableVmCreate(edited)

Comment (upalatucci) [reply src/views/catalog/wizard/tabs/environment/WizardEnvironmentTab.tsx:25]: i can event pass only `setDisableVMCreate` :-) Thanks for the suggestion

File: src/utils/components/EnvironmentEditor/EnvironmentForm.tsx (hunk 1 above)
Comment (upalatucci) [reply src/utils/components/EnvironmentEditor/EnvironmentForm.tsx]: yes. okay

**Review by pcbailey — COMMENTED** [review]
One small nitpick. Definitely not worth holding it up.
/lgtm

File: src/utils/components/EnvironmentEditor/EnvironmentForm.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+      <div className="row">
+        <div className="col-xs-12">
+          <Button
+            className="pf-m-link--align-left"
+            onClick={onEnvironmentAdd}
+            type="button"
+            variant="link"
+          >
+            <PlusCircleIcon /> {t('Add Config Map, Secret or Service Account')}
```
[the remark this answers is not in the document]
Comment (upalatucci) [reply src/utils/components/EnvironmentEditor/EnvironmentForm.tsx:88]: Okay. Thanks for not holding this. I'll fix in the next pr

**Pull request merged**